### PR TITLE
Generic Service State Machine - added notifications and improved logging.

### PR DESCRIPTION
**Pull request by tinaafitz**
Added notifications, improved logging and updated provision instance for finished state.

### CODE REVIEW

File: content/automate/ManageIQ/Service/Generic/StateMachines/GenericLifecycle.class/provision.yaml [hunk 1]
```diff
@@ -48,3 +48,4 @@ object:
       on_error: update_status(status => 'Error Emailing Owner')
   - finish:
       value: METHOD::finish
+      on_exit: update_status(status => "${/#service_action} Completed Successfully")
```
Comment (mkanoor) [inline content/automate/ManageIQ/Service/Generic/StateMachines/GenericLifecycle.class/provision.yaml]: @tinaafitz 
The service action would be Provision|Reconfigure|Retire can we map that to status?

File: spec/content/automate/ManageIQ/Service/Generic/StateMachines/GenericLifecycle.class/__methods__/finish_spec.rb [hunk 2]
```diff
@@ -9,17 +9,18 @@
...
   context "finish" do
     it "check success" do
       allow(svc_task).to receive(:destination).and_return(svc_service)
       allow(svc_service).to receive(:main).and_return(nil)
+      allow(ae_service).to receive(:create_notification)
 
       described_class.new(ae_service).main
 
-      expect(ae_service.root['ae_result']).to eq("ok")
+      expect(ae_service.root['ae_result']).to be_nil
```
Comment (mkanoor) [inline spec/content/automate/ManageIQ/Service/Generic/StateMachines/GenericLifecycle.class/__methods__/finish_spec.rb]: @tinaafitz 
we are checking the value to be nil but no one is setting it. A better check would be to set the ae_result to ok and check for that.

File: spec/content/automate/ManageIQ/Service/Generic/StateMachines/GenericLifecycle.class/__methods__/finish_spec.rb [hunk 3]
```diff
@@ -9,17 +9,18 @@
...
   let(:svc_task) { MiqAeMethodService::MiqAeServiceServiceTemplateProvisionTask.find(task.id) }
   let(:svc_service) { MiqAeMethodService::MiqAeServiceServiceAnsibleTower.find(service_ansible_tower.id) }
-  let(:root_object) { Spec::Support::MiqAeMockObject.new('service_template_provision_task' => svc_task) }
+  let(:root_object) { Spec::Support::MiqAeMockObject.new('service' => svc_service, 'service_action' => 'Provision') }
   let(:ae_service) { Spec::Support::MiqAeMockService.new(root_object) }
 
   context "finish" do
     it "check success" do
       allow(svc_task).to receive(:destination).and_return(svc_service)
       allow(svc_service).to receive(:main).and_return(nil)
```
Comment (mkanoor) [inline spec/content/automate/ManageIQ/Service/Generic/StateMachines/GenericLifecycle.class/__methods__/finish_spec.rb]: @tinaafitz 
Is there a main method for service model service?

File: content/automate/ManageIQ/Service/Generic/StateMachines/GenericLifecycle.class/provision.yaml (hunk 1 above)
Comment (tinaafitz) [inline content/automate/ManageIQ/Service/Generic/StateMachines/GenericLifecycle.class/provision.yaml]: @mkanoor Status here is a message. Will create a followup PR to rename status to message.

**Comment by tinaafitz** [comment]
@mkanoor Changes made.

**Pull request merged**